### DISCUSSION TITLE
Next.js 13 webpack error - but Next.js 13 doesn't use webpack anymore?

### DISCUSSION THREAD

**Discussion by richleach** [vercel/next.js#47672] [Help]
### Summary

I installed and setup a Next.js 13 project and I'm trying "npm run dev" but when I do I get a server console error: 

[[webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies]

I'm wondering if this is even a valid error as it's my understanding that Next.js 13 doesn't even use webpack anymore does it? I thought 13 used Rust instead.... Is webpack still used in other parts of a Next.js 13?

I'm trying to work through this error, I'm on a Mac OS Big Sur 11.7.4
Node.js v18.15.0
npm 9.5.0

From my package.json file:
"dependencies": {
    "@types/node": "18.15.11",
    "@types/react": "18.0.31",
    "@types/react-dom": "18.0.11",
    "next": "13.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "5.0.2"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.14",
    "postcss": "^8.4.21",
    "tailwindcss": "^3.3.0"
  }

Thanks in advance for your help.

Rich

**Comment by icyJoseph** [comment]
Hi,

Next.js uses SWC, but still bundles with Webpack through `swc-loader`, or `next-swc-loader` to be precise.

I think you can play this extreme simplification:

- Vite = Rollup + esbuild (or swc)
- Next.js = Webpack + SWC

Then you could understand why someone would ask this question, https://github.com/vitejs/vite/discussions/7622 which would take you here:

- https://vitejs.dev/guide/why.html#why-not-bundle-with-esbuild

In Next.js' case, the situation is somewhat similar, but with Webpack, for example in the `next.config.js` file you can tap into the webpack configuration:

- https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config

What the Vercel team is trying to do now, is to move into TurboPack: https://turbo.build/pack, but that's still an on-going development.

A piece of trivia. Next.js uses compiled dependencies. If you do `yarn why webpack` or `npm explain webpack` you'd see something like (assuming no other packages like Storybook v6 are used):

```console
yarn why webpack
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "webpack"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 0.08s.
```

That might lead us to think that indeed, Webpack is not used, but that's not the case. What is really happening is that Next.js has pre-compiled Webpack, and a bunch of other libraries, which you can see here:

- https://github.com/vercel/next.js/tree/canary/packages/next/src/compiled

Why is your dev process failing, I can't tell with this much info, just yet...

**Reply by richleach** [reply]
Thanks so much for you in-depth explanation, I really appreciate it and understand the issue so much better now.

Unfortunately I'm still not able to get passed this error message, and I think that (at least, in part) it's responsible for preventing me from deploying my project to my host (Vercel).... I'm going to continue working this error, and thank you again!

**Reply by icyJoseph** [reply]
Yeah I am afraid this issue seems very obscure.

You might have found: https://github.com/vercel/next.js/discussions/35877#discussioncomment-2501162, which doesn't apply here, because you are on MacOS.

- Have you tried starting a new project? Like in another directory?
- Have you tried reinstalling Node?
- Maybe you have webpack installed globally?

**Reply by richleach** [reply]
As a test on machine 1 (Node v 18.15.0 and NPM v 9.5.0)(where I've been doing this development work) I started a new Next project in a new directory (the base Next.js site only, I didn't add or modify any of the code after I installed it onto my machine.
When I run "npm run dev" I still get the error:

<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies

I have the good fortune of a second Mac in my office (machine 2 Node v 16.16.0 and NPM v 8.11.0) and I do NOT get any webpack errors when running "npm run dev".

So clearly, at the very least the version(s) of Node and/or NPM are affecting this issue (assuming all other conditions are equal).

If I reinstall Node should I aim to use the latest stable version? Clearly the old(er) version seems to be working for me in this case....
And is having webpack installed globally not a good idea? It seems that I do, in fact have webpack installed globally.

Thanks so much for your help I really appreciate your insight.

**Reply by richleach** [reply]
I never could get past this error, I had to rebuild everything on a completely different install/machine.... Wish I had more time to devote to this issue, would've liked to have helped more people
